jmbase/jmbase/commands.py

```diff
@@ -182,7 +182,8 @@ class JMOffers(JMCommand):
     """Return the entire contents of the
     orderbook to TAKER, as a json-ified dict.
     """
-    arguments = [(b'orderbook', BigUnicode())]
+    arguments = [(b'orderbook', BigUnicode()),
+                 (b'fidelitybonds', BigUnicode())]
```

Comment:
Orderbook is a dict keyed by counterparty; wouldn't it make more sense to append the fidelity bond info within the same dict with those keys rather than have a new one? (I may revise this comment after reading more code!)

Reply:
I think no because a single maker can advertise multiple offers, and each of those offer dicts would have to have fidelity bond information redundantly included.

Reply:
Yeah I was not remembering right here; orderbooks passed over the wire look like:

```
[{'counterparty': 'J5DnSfWe6jnRdVap', 'oid': 0, 'ordertype': 'sw0reloffer',
'minsize': 7499999, 'maxsize': 599972700, 'txfee': 100, 'cjfee': '0.00002'},
{'counterparty': 'J5CmrfAkaWKD532O', 'oid': 0, 'ordertype': 'sw0reloffer',
'minsize': 7499999, 'maxsize': 599972700, 'txfee': 100, 'cjfee': '0.00002'},
{'counterparty': 'J58JcrGfmX87RH8w', 'oid': 0, 'ordertype': 'sw0reloffer',
'minsize': 7499999, 'maxsize': 599972700, 'txfee': 100, 'cjfee': '0.00002'}]
```

so it's a list of dicts keyed by `'counterparty'`, not a dict keyed by counterparty :) To be honest I do still think it'd be better to reorganize it into:
list(dict(key: "J5...", value: dict(key: "fidelitybond", value: "..", key: "orders", value: list(order1dict, order2dict..))).
... or something similar.

However this would be a big change. So I think your choice is reasonable to avoid changing the existing structure.